kart/tabular/working_copy/postgis.py

```diff
@@ -337,3 +341,20 @@ def _apply_meta_schema_json(self, sess, dataset, src_value, dest_value):
             sess.execute(
                 f"""ALTER TABLE {self.table_identifier(table)} ALTER COLUMN {self.quote(col.name)} TYPE {dest_type};"""
             )
+
+    def _check_for_unsupported_ds_types(self, sess, target_datasets):
+        """Check for spatial datasets and verify if the PostGIS extension is enabled in the database."""
+        for dataset_path, dataset in target_datasets.items():
+            if dataset.is_spatial:
+                spatial_ref_sys_exists = sess.scalar(
+                    """
+                    SELECT COUNT(*) FROM information_schema.tables
+                    WHERE table_schema='public' AND table_name='spatial_ref_sys';
+                    """
+                )
+                if not spatial_ref_sys_exists:
```

Comment:
We can early return here if spatial_ref_sys_exists is True - if spatial_ref_sys_exists is True then there's no reason to keep checking further, since we know we'll never hit the error condition